File: app/src/main/java/org/neidhardt/dynamicsoundboard/navigationdrawer/views/NavigationDrawerList.java
package org.neidhardt.dynamicsoundboard.navigationdrawer.views;

import android.content.Context;
import android.util.AttributeSet;
import android.widget.FrameLayout;


public abstract class NavigationDrawerList extends FrameLayout
{
	public NavigationDrawerList(Context context)
	{
		super(context);
	}

	public NavigationDrawerList(Context context, AttributeSet attrs)
	{
		super(context, attrs);
	}

	public NavigationDrawerList(Context context, AttributeSet attrs, int defStyleAttr)
	{
		super(context, attrs, defStyleAttr);
	}

	protected abstract int getItemCount();

	protected abstract int getActionModeTitle();

	public abstract NavigationDrawerListPresenter getPresenter();
}

File: app/src/main/kotlin/org/neidhardt/dynamicsoundboard/navigationdrawer/views/NavigationDrawerList.kt
package org.neidhardt.dynamicsoundboard.navigationdrawer.views

import android.content.Context
import android.util.AttributeSet
import android.widget.FrameLayout


abstract class NavigationDrawerList : FrameLayout
{
	constructor(context: Context) : super(context)

	constructor(context: Context, attrs: AttributeSet) : super(context, attrs)

	constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int) : super(context, attrs, defStyleAttr)

	abstract val itemCount: Int

	abstract val actionModeTitle: Int

}
